esp-wifi/src/wifi/mod.rs

```diff
@@ -593,18 +576,22 @@ pub struct EapClientConfiguration {
     pub auth_method: AuthMethod,
 
     /// The identity used during authentication.
-    pub identity: Option<heapless::String<128>>,
+    #[cfg_attr(feature = "defmt", defmt(Debug2Format))]
```

Comment:
This shouldn't be needed 👀

Reply:
it unfortunately is - either that or enabling the `alloc` feature on `defmt` which causes compilation errors for ESP32-S2 (why not ESP32-C2/ESP32-C3 - ah ok I see: https://github.com/knurling-rs/defmt/blob/aad3615993d0aaf891bf7fe9f1a6188cdc435dcb/defmt/src/impls/alloc_.rs#L17-L23 ???)

Reply:
Maybe at some point we should create a PR to update: https://github.com/knurling-rs/defmt/blob/aad3615993d0aaf891bf7fe9f1a6188cdc435dcb/defmt/build.rs#L18-L30

Reply:
Right now is a good time, if it's as simple as adding `xtensa-esp32s2-none-elf` to that list :)

Reply:
https://github.com/knurling-rs/defmt/pull/955 I will test run this to see if it's enough, then mark the PR ready or add what else is needed. In the mean time, I guess we should be fine with this horrible set of annotations.